pkg/api/alert/v1/types.go

```diff
@@ -60,9 +60,9 @@ type AlertSpec struct {
 	AlertMemory string `json:"alertMemory,omitempty"`
 	CfsslMemory string `json:"cfsslMemory,omitempty"`
 
-	DesiredState          string                    `json:"desiredState,omitempty"`
-	ImageRegistries       []string                  `json:"imageRegistries,omitempty"`
-	RegistryConfiguration api.RegistryConfiguration `json:"registryConfiguration,omitempty"`
+	DesiredState          string                     `json:"desiredState,omitempty"`
+	ImageRegistries       []string                   `json:"imageRegistries,omitempty"`
+	RegistryConfiguration *api.RegistryConfiguration `json:"registryConfiguration,omitempty"`
```

Comment:
If we change RegistryConfiguration to a pointer in the Alert Spec then we need to check if this field is nil inside ctl_alert.go  before assigning registry / namespace / pull secrets.

Reply:
I think that should be fine right..